Dockerfile

```diff
@@ -8,16 +8,30 @@ COPY hash_cn /hash_cn
 RUN sed -ri "s/^(.*DonationLevel = )[0-9]\.[0-9]{2}/\1${DONATION_LEVEL}/" /server/Server/DevDonation.cs && \
 	apt-get -qq update && \
 	apt-get -qq install build-essential && \
+	rm -rf /var/lib/apt/lists/* && \
 	cd /hash_cn/libhash && \
 	make && \
 	cd /server && \
 	msbuild Server.sln /p:Configuration=Release_Server /p:Platform="any CPU"
 
-FROM mono:5.12.0.226
-
-VOLUME ["/root"]
+FROM mono:5.16
 
 RUN mkdir /webminerpool
+
+# Install acme.sh
+RUN apt-get -qq update && \
+	apt-get install -qq \
+		coreutils \
+		cron \
+		curl \
+		git \
+		openssl \
+		socat && \
+	rm -rf /var/lib/apt/lists/* && \
+	git clone https://github.com/Neilpang/acme.sh.git /root/acme.sh && \
+	cd /root/acme.sh && \
+	git checkout 2.7.9 && \
```

Comment:
Please externalize version to `ENV` on next update and, ideally, pull from release tarball as opposed to using Git. Here's how you can pull release tarball:

- Using `latest` version meta from NPM: https://git.habd.as/comfusion/after-dark/src/commit/de243545c422d4190e441ffe8293bbbf5dfb4d4e/bin/install#L34-L38
- Using `wget` from GitHub _without_ installing `git`: https://git.habd.as/jhabdas/react-native-webpack-starter-kit/src/commit/6a19524e987052bb1bc171bdcaaa8664d43c946b/Dockerfile#L21-L26

The benefit of not using `git` is that you may splitting up your containers and use `busybox` for the basics once the binary is compiled in a multi-image build.